File: Backend/Models/Client/UserModel.cs
using Backend.Models.Backend;

namespace Backend.Models.Client
{
    public class UserModel
    {
        public String Id { get; set; }
        public String Email { get; set; }
        public String Login { get; set; }
        public String Password { get; set; }

        public UserModel(String id, String email, String login, String password)
        {
            Id = id;
            Email = email;
            Login = login;
            Password = password;
        }

        public static UserModel From(User user)
        {
            return new UserModel(user.Id, user.Email, user.Login, user.Password);
        }
    }
}

File: Client/src/app/components/models/user.model.ts
export class UserModel {
    
    public id: string;
    public login: string;
    public email: string;
    public password: string;

    constructor(data: IUserModel) {
        this.id = data.id;
        this.login = data.login;
        this.email = data.email;
        this.password = data.password;
    }

}

export interface IUserModel {
    id: string;
    login: string;
    email: string;
    password: string;
}